core/src/main/java/org/mobilitydata/gtfsvalidator/input/GtfsInput.java

```diff
@@ -44,6 +54,78 @@ static GtfsInput createFromPath(String path) throws IOException {
         return new GtfsZipFileInput(file);
     }
 
+    /**
+     * Creates an specific GtfsInput to read data from the given URL.
+     *
+     * @param sourceUrl
+     * @param targetPathAsString
+     * @return
+     * @throws IOException
+     */
+    static GtfsInput createFromUrl(URL sourceUrl, String targetPathAsString) throws IOException {
+        InputStream inputStream;
+
+        try {
+            final HttpURLConnection httpConnection = (HttpURLConnection) sourceUrl.openConnection();
+            final int responseCode = httpConnection.getResponseCode();
+            // check response code
+            if (responseCode == HTTP_MOVED_PERM || responseCode == HTTP_MOVED_TEMP ||
+                    responseCode == HTTP_TEMP_REDIRECT || responseCode == HTTP_PERM_REDIRECT) {
+                // use redirection instead of original url
+                final String newUrlAsString = httpConnection.getHeaderField("Location");
+                final URLConnection connection = new URL(newUrlAsString).openConnection();
+                inputStream = connection.getInputStream();
+            } else {
+                inputStream = sourceUrl.openStream();
+            }
+            Path path = createPath(targetPathAsString);
+            Files.copy(
+                    inputStream,
+                    path,
+                    StandardCopyOption.REPLACE_EXISTING
+            );
+            return createFromPath(path.toString());
+        } catch (IOException e) {
+            throw e;
+        }
+    }
+
+    /**
+     * Creates a path from a given string
+     * @param toCleanOrCreate
+     * @return
+     */
+    static Path createPath(final String toCleanOrCreate) {
+        // to empty any already existing directory
+        Path pathToCleanOrCreate = Paths.get(toCleanOrCreate);
+        if (Files.exists(pathToCleanOrCreate)) {
+            try {
+                //noinspection ResultOfMethodCallIgnored -- we ignore if deletion went well or not
+                Files.walk(pathToCleanOrCreate).sorted(Comparator.reverseOrder()).map(Path::toFile).forEach(File::delete);
```

Comment:
I suggest org.apache.commons.io.FileUtils.deleteDirectory that deletes directories recursively.

Reply:
🙏

Reply:
I actually realize that this piece of code needs to be re-worked: we just have to create/delete the `.zip` file that will contain the zipped dataset after download. If you agree I will use `org.apache.commons.io.FileUtils.deleteQuietly` which deletes files instead of directories.